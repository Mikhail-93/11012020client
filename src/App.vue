<template>
    <v-app id="inspire">
        <v-navigation-drawer v-model="drawer" app>
            <v-list dense>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-home</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Home</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-contact-mail</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Contact</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
            </v-list>
        </v-navigation-drawer>

        <v-app-bar
                app
                color="indigo"
                dark
        >
            <v-app-bar-nav-icon @click.stop="drawer = !drawer"/>
            <v-toolbar-title>Application</v-toolbar-title>
        </v-app-bar>

        <v-content>
            <v-container class="fill-height pa-0" fluid>
                <LMap ref="map" style="z-index: 0">
                    <l-tile-layer
                            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
                            attribution='Map data: &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>, <a href="http://viewfinderpanoramas.org">SRTM</a> | Map style: &copy; <a href="https://opentopomap.org">OpenTopoMap</a> (<a href="https://creativecommons.org/licenses/by-sa/3.0/">CC-BY-SA</a>)'
                    >
                    </l-tile-layer>
                    <!--                    <l-control-attribution position="bottomright" prefix="leaflet OpenStreetMap Contributions" >-->

                </LMap>
                <!--                class="align-stretch"-->
            </v-container>
        </v-content>
    </v-app>
</template>

<script>
    import {LMap, LTileLayer, LMarker, LControlAttribution} from 'vue2-leaflet';

    export default {
        components: {
            LMap,
            LTileLayer,
            LMarker,
            LControlAttribution
        },
        props: {
            source: String,
        },
        data: () => ({
            drawer: null,
        }),
        mounted() {
            this.$nextTick(() => {
                // this.map=
                this.$refs.map.mapObject.setView([51.505, -0.09], 13)

            })
        }
    }
</script>