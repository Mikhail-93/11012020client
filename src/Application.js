import {AbstractSync, Kopnik} from "./models";

export default class Application {
    static getInstance() {
        if (!this.instance) {
            this.instance = new this()
        }
        return this.instance
    }

    user = null

    constructor() {
        /**
         * Кэш моделей
         * @type {Array}
         */
        // пока не понятно зачем приложению кэш.
        // он вероятно будет сильно торможить приложение из-за реактивности App.$data в которой он находится
        // хотя он же не в App.$data а в App.$options !!
        // уже в дата потому тчо в options не реактивен
        //this.models = AbstractSync.cache

        /**
         * Идентификатор раздела
         * Равен названиею соответствующего компонента
         *
         * @type {string} Map | Profile | Thanks
         */
        this.SECTION = "Map"

        this.user = undefined

        this.initUser()
    }

    getSharedState() {
        return {
            SECTION: this.SECTION
        }
    }

    setState(state) {
        if (state.SECTION) {
            this.SECTION = state.SECTION
        }
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    async initUser() {
        let vkUser = localStorage.getItem("vkUser")
        if (vkUser) {
            vkUser = JSON.parse(vkUser)
            let user //= await Kopnik.getVkUserStatus(vkUser.uid)
            if (vkUserStatus) {
                this.user= Kopnik.merge(vkUserStatus.user)
            } else {
                this.setVkUser(vkUser)
                this.SECTION="Profile"
            }
        }
    }

    /**
     * Учстанавливает пользователем объект из VK.init::success()
     * @param vkUser
     * @returns
     */
    setVkUser(vkUser){
        this.user= new Kopnik
        this.user.merge({
            hash: vkUser.hash,
            uid: vkUser.uid,
            firstname: vkUser.first_name,
            surname: vkUser.last_name,
            photo: vkUser.photo,
            smallPhoto: vkUser.photo_rec,
        })
        localStorage.setItem("vkUser", JSON.stringify(vkUser))
    }
}
