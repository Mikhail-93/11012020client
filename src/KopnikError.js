class KopnikError extends Error {
    constructor(message, code) {
        super(message)
        this.code = code
    }
}
class KopnikApiError extends KopnikError {
    constructor(message, code, url) {
        super(message, code)
        this.url = url
    }
}



export {KopnikError, KopnikApiError}
