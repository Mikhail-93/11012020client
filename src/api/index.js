import 'isomorphic-fetch'
import fetchIntercept from "fetch-intercept/lib/node";

import errorInterceptor from "./errorInterceptor";
import dataInterceptor from "./dataInterceptor";
import cookieInterceptor from "./cookieInterceptor";
import headersInterceptor from "./headersInterceptor";
import api from "./api";

fetchIntercept.register(errorInterceptor)
fetchIntercept.register(dataInterceptor)
fetchIntercept.register(cookieInterceptor)
fetchIntercept.register(headersInterceptor)



export default api
