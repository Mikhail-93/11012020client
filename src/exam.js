const cars = {};Debugger attached.
const carHonda = {model: 'CR-V', year: 2015};
const carVolvo = {model: 'XC-60', year: 2018};
const carBMW = {model: 'X5', year: 2019};
cars[carHonda] = 'Good';
cars[carVolvo] = 'Better';
cars[carBMW] = 'The best';
console.log('Honda is', cars[carHonda]);
console.log('Volvo is', cars[carVolvo]);
console.log('BMW is', cars[carBMW]);
