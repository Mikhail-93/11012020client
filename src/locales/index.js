import ru from './ru'
import en from './en'
import sk from './sk.json'

const messages = {
  ru,
  en,
  sk,
}

export default messages
