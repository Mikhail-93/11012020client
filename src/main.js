import Vue from 'vue'
import vuetify from './plugins/vuetify'
import router from './plugins/vue-router'
import i18n from './plugins/i18n'
import "./plugins/vee-validate"

import {Icon} from 'leaflet'
import 'leaflet/dist/leaflet.css'
// this part resolve an issue where the markers would not appear
delete Icon.Default.prototype._getIconUrl;
Icon.Default.mergeOptions({
    iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
    iconUrl: require('leaflet/dist/images/marker-icon.png'),
    shadowUrl: require('leaflet/dist/images/marker-shadow.png')
})

import Application from "./Application";
import App from './components/AppVue.vue'

global.app = new Application

Vue.config.productionTip = false

global.vue = new Vue({
    ...App,
    el: "#appContainer",
    vuetify,
    i18n,
    router,
    // render: h => h(App)
})//.$mount('#app')


