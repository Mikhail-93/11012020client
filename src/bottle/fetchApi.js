import {container} from './bottle'
import _ from 'lodash'
import jsonToFormData from 'json-form-data'
import {KopnikApiError} from "../KopnikError";


export default async function fetchApi(url, options = {}) {
    const defaultOptions = {
            credentials: 'include',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'text/plain',
            }
        },
        cookieOptions = (container.config.di.cookie && container.cookieService.cookie) ? {headers: {Cookie: container.cookieService.cookie}} : null

    options = _.merge({}, defaultOptions, cookieOptions, options)
    if (options.body) {
        options.body = JSON.stringify(options.body)
    }
    // container.logger.warn(options)
    // console.log(container.config.api.path)
    let fullUrl = `${container.config.api.path}/${url}`.replace(/\w+\/\.\.\//, '')

    try {
        // console.log(url, options)
        var response = await fetch(fullUrl, options)
        // Пропал 4G
    } catch (err) {
        throw new KopnikApiError(err.message, null, fullUrl)
    }
    // Не найдена страница или синтаксическая ошибка в веб-сервисе
    if (!response.ok) {
        throw new KopnikApiError(response.statusText, response.status, fullUrl)
    }

    let cookie = response.headers.get('set-cookie')
    if (container.config.di.cookie && cookie) {
        cookie = cookie.match(/(\w+=(\w|\d)+)/)[0]
        container.cookieService.cookie = cookie
    }
    // container.logger.warn(cookie, response.headers)
    let result
    switch (response.headers.get('Content-Type')) {
        case 'application/json':
            // result = await response.text()
            result = await response.json()
            // Не авторизован/Нет такого пользователя
            if (result.error) {
                console.log(result.error)
                throw new KopnikApiError(result.error.error_msg, result.error.error_code, fullUrl)
            }
            break
        default:
            result = await response.text()
    }

    return result.response
}
