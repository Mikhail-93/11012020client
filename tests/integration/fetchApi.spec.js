import {KopnikApiError} from "../../src/KopnikError";

require('isomorphic-fetch');
import prettyFormat from 'pretty-format'

import fetchApi from "../../src/bottle/fetchApi";

describe('integration/fetchApi', () => {
    it('users/get?ids=', async () => {
        let result = await fetchApi('users/get?ids=')
        expect(result).toMatchSnapshot()
    });
    it('users/get?ids=1', async () => {
        let result = await fetchApi('users/get?ids=1')
        expect(result).toMatchSnapshot()
    });
    it('users/get?ids=2', async () => {
        let result = await fetchApi('users/get?ids=2')
        expect(result).toMatchSnapshot()
    });
    it('users/get?ids=3', async () => {
        let result = await fetchApi('users/get?ids=3')
        expect(result).toMatchSnapshot()
    });
    it('users/get?ids=4', async () => {
        let result = await fetchApi('users/get?ids=4')
        expect(result).toMatchSnapshot()
    });
    it('users/get?ids=1,2,3', async () => {
        let result = await fetchApi('users/get?ids=1,2,3')
        expect(result).toMatchSnapshot()
    })
})

