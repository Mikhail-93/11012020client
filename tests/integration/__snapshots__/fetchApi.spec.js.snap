// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration.fetchApi users/get?ids= 1`] = `
Array [
  Object {
    "birthyear": 1231,
    "firstName": "Алексей",
    "foreman_id": null,
    "id": 2,
    "lastName": "Баранов",
    "location": Array [
      55.755826,
      37.6173,
    ],
    "passport": 1231,
    "patronymic": "qwer",
    "photo": "@todo",
    "smallPhoto": "@todo",
    "status": 2,
    "witness_id": null,
  },
]
`;

exports[`integration.fetchApi users/get?ids=1,2,3 1`] = `
Array [
  Object {
    "birthyear": 1234,
    "firstName": "Арт",
    "foreman_id": null,
    "id": 1,
    "lastName": "Арт",
    "location": Array [
      55.030494139179,
      82.918603420258,
    ],
    "passport": null,
    "patronymic": "Арт",
    "photo": "@todo",
    "smallPhoto": "@todo",
    "status": 2,
    "witness_id": 2,
  },
  Object {
    "birthyear": 1231,
    "firstName": "Алексей",
    "foreman_id": null,
    "id": 2,
    "lastName": "Баранов",
    "location": Array [
      55.755826,
      37.6173,
    ],
    "passport": 1231,
    "patronymic": "qwer",
    "photo": "@todo",
    "smallPhoto": "@todo",
    "status": 2,
    "witness_id": null,
  },
  Object {
    "birthyear": 1982,
    "firstName": "Александр",
    "foreman_id": null,
    "id": 3,
    "lastName": "Титов",
    "location": Array [
      55.158396343962,
      49.176560640553,
    ],
    "passport": null,
    "patronymic": "Николаевич",
    "photo": "@todo",
    "smallPhoto": "@todo",
    "status": 2,
    "witness_id": 2,
  },
]
`;

exports[`integration.fetchApi users/get?ids=2 1`] = `
Array [
  Object {
    "birthyear": 1231,
    "firstName": "Алексей",
    "foreman_id": null,
    "id": 2,
    "lastName": "Баранов",
    "location": Array [
      55.755826,
      37.6173,
    ],
    "passport": 1231,
    "patronymic": "qwer",
    "photo": "@todo",
    "smallPhoto": "@todo",
    "status": 2,
    "witness_id": null,
  },
]
`;
